shuffle-plugin/src/main/scala/com/nvidia/spark/rapids/shuffle/ucx/UCX.scala

```diff
@@ -350,13 +383,25 @@ class UCX(executorId: Int, usingWakeupFeature: Boolean = true) extends AutoClose
   private[ucx] def setupEndpoint(
       endpointId: Long, workerAddress: WorkerAddress, peerRkeys: Rkeys): UcpEndpoint = {
     logDebug(s"Starting/reusing an endpoint to $workerAddress with id $endpointId")
-    // create an UCX endpoint using workerAddress
+    // create an UCX endpoint using workerAddress or socket address
     endpoints.computeIfAbsent(endpointId,
       (_: Long) => {
         logInfo(s"No endpoint found for $endpointId. Adding it.")
-        val ep = worker.newEndpoint(
-          new UcpEndpointParams()
-            .setUcpAddress(workerAddress.address))
+        val epParams =  new UcpEndpointParams().setPeerErrorHandlingMode().setErrorHandler(
+          (ucpEndpoint: UcpEndpoint, i: Int, errorString: String) => {
+            logError(s"Endpoint to $endpointId got error: $errorString")
```

Comment:
Need to check the status field here given our offline conversation, and output error if it's truly an error, as this callback is called also when the endpoint is regularly closed.